### PR TITLE
CLOUDP-93929: fix mongocli deb and rpm packaging

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-93929

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

goreleaser changed how they package debs and rpms, this is the second time our debs and rpms break so adding some postpackaging testing here so it doesn't happen again

### CODE REVIEW

File: build/ci/evergreen.yml [hunk 1]
```diff
@@ -861,49 +870,6 @@ tasks:
       - func: "e2e test"
         vars:
           E2E_TAGS: opsmanager,remote,sharded
-  # Deploy a replica set, we need to be careful on running parallel modifications to the automation config
```
Comment (gssbzn) [inline build/ci/evergreen.yml:864]: not going to happen given how close we are to EOL

File: build/ci/evergreen.yml [hunk 2]
```diff
@@ -392,7 +402,6 @@ functions:
           - src/github.com/mongodb/mongocli/dist/*.rpm
           - src/github.com/mongodb/mongocli/dist/*.tgz
           - src/github.com/mongodb/mongocli/dist/*.json
-          - src/github.com/mongodb/mongocli/dist/*.rb
```
Comment (gssbzn) [inline build/ci/evergreen.yml:395]: we no longer publish on brew

File: build/ci/evergreen.yml (hunk 1 above)
Comment (colm-quinn) [reply build/ci/evergreen.yml:864]: [q] 4.2 tests are failing, is there a plan for a 4.4 version of the same test or it's no longer necessary?

File: build/package/centos7-rpm.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+FROM centos:7
```
Comment (colm-quinn) [inline build/package/centos7-rpm.Dockerfile:1]: How I wish we could pass the `FROM` in as a variable :/

**Review by colm-quinn — APPROVED** [review]
Great work here getting all those configurations working! LGTM

File: build/ci/evergreen.yml (hunk 1 above)
Comment (gssbzn) [reply build/ci/evergreen.yml:864]: we never enabled them as there was an automaton test failing due to changes between 4.2 and 4.4
We already have OM4.4 tests in place TBD 5.0 tests but we have cloud manager tests which at this point they are almost the same

Comment (gssbzn) [reply build/ci/evergreen.yml:864]: also mongocli was always a 4.4 first 4.2 best effort (given timings between the launch of mcli and 4.4(

Comment (colm-quinn) [reply build/ci/evergreen.yml:864]: Cool, thanks for the context!

File: build/package/centos7-rpm.Dockerfile (hunk 3 above)
Comment (gssbzn) [reply build/package/centos7-rpm.Dockerfile:1]: yep, I know there's some tooling to have templates but felt maybe overkill for now

Comment (colm-quinn) [reply build/package/centos7-rpm.Dockerfile:1]: Yup agreed

**Review by andreaangiolillo — APPROVED** [review]
LGTM 💯

**Pull request merged**